tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserRegistry.java

```diff
@@ -578,6 +582,43 @@ public CLDRLocale exampleLocale() {
             }
             return null;
         }
+
+        public void signCla(ClaSignature cla) {
+            if (ClaSignature.CLA_ORGS.contains(getOrganization())) {
+                throw new IllegalArgumentException("Cannot modify CLA for a listed org");
+            }
+            if (!cla.valid()) throw new IllegalArgumentException("Invalid CLA");
+            cla.signed = new Date();
+            settings().setJson(ClaSignature.CLA_KEY, cla);
+            claSigned = true;
+        }
+
+        public boolean revokeCla() {
+            if (ClaSignature.CLA_ORGS.contains(getOrganization())) {
+                return false;
+            }
+            final String oldCla = settings().get(ClaSignature.CLA_KEY, null);
+            if (oldCla == null || oldCla.isBlank()) {
+                return false;
+            }
+            settings().set(ClaSignature.CLA_KEY, "");
+            claSigned = false;
+            return true;
+        }
+
+        public ClaSignature getCla() {
+            CLDRConfig config = CLDRConfig.getInstance();
+            if (config.getEnvironment() == Environment.UNITTEST) {
+                return new ClaSignature("UNITTEST");
+            } else if (ClaSignature.DO_NOT_REQURE_CLA
+                    && !config.getProperty("REQUIRE_CLA", false)) {
+                // no CLA needed unless CLDR_NEED_CLA is true.
+                return new ClaSignature("DO_NOT_REQURE_CLA");
```

Comment:
this constructor presumably sets signed = true or signed = Date... OK, I've confirmed this suspicion